src/vs/workbench/parts/terminal/electron-browser/terminalConfigHelper.ts

```diff
@@ -67,34 +69,73 @@ const DEFAULT_ANSI_COLORS = {
 	]
 };
 
+export interface ITerminalFont {
+	fontFamily: string;
+	fontSize: number;
+	lineHeight: number;
+	charWidth: number;
+	charHeight: number;
+}
+
 /**
  * Encapsulates terminal configuration logic, the primary purpose of this file is so that platform
  * specific test cases can be written.
  */
 export class TerminalConfigHelper {
+	private charMeasureElement: HTMLElement;
+
 	public constructor(
 		private platform: Platform,
 		private configurationService: IConfigurationService,
-		private themeService: IThemeService) {
+		private themeService: IThemeService,
+		private parentDomElement: HTMLElement) {
 	}
 
 	public getTheme(): string[] {
 		let baseThemeId = getBaseThemeId(this.themeService.getTheme());
 		return DEFAULT_ANSI_COLORS[baseThemeId];
 	}
 
+	private neasureFont(fontFamily: string, fontSize: number, lineHeight: number): ITerminalFont {
+		if (!this.charMeasureElement) {
+			this.charMeasureElement = new Builder(this.parentDomElement, true).div().build().getHTMLElement();
+		}
+		let style = this.charMeasureElement.style;
+		style.display = 'inline';
```

Comment:
I'll check, but this triggers a synchronous layout and then immediatelly removes it I think it is not visible.